### PR TITLE
plugin,metadata: avoid specifying API version

### PR DESCRIPTION
**Pull request by mmilata**
Even though osbs uses v1beta3 now the plugin had hardcoded configuration
for v1beta1.

### CODE REVIEW

File: atomic_reactor/plugins/post_store_metadata_in_osv3.py [hunk 1]
```diff
@@ -55,13 +58,23 @@ def run(self):
...
         self.log.info("build id = %s", build_id)
 
-        api_url = urljoin(self.url, "/osapi/v1beta1/")
-        oauth_url = urljoin(self.url, "/oauth/authorize")  # MUST NOT END WITH SLASH
-
         # initial setup will use host based auth: apache will be set to accept everything
         # from specific IP and will set specific X-Remote-User for such requests
-        # FIXME: use OSBS here
-        o = Openshift(api_url, oauth_url, None, use_auth=self.use_auth, verify_ssl=self.verify_ssl)
+        with tempfile.NamedTemporaryFile(mode='w+') as f:
```
Comment (TomasTomecek) [inline atomic_reactor/plugins/post_store_metadata_in_osv3.py]: I think we don't need to store anything on disk to make this work, something like this should be possible:
```
Configuration(conf_file=None, conf_section=None, cli_args=None, openshift_uri="...", use_auth=..., ver...)
```
If this won't work, I'm in favor of patching osbs.

File: atomic_reactor/plugins/post_store_metadata_in_osv3.py [hunk 2]
```diff
@@ -111,4 +124,4 @@ def run(self):
                 "sha256sum": tar_sha256sum,
                 "filename": os.path.basename(tar_path),
             })
-        o.set_annotations_on_build(build_id, labels)
+        osbs.os.set_annotations_on_build(build_id, labels)
```
Comment (TomasTomecek) [inline atomic_reactor/plugins/post_store_metadata_in_osv3.py]: hm, looks like we should make api method for annotations

**Comment by mmilata** [comment]
I'll fix OSBS to not require a config file (conf_file=None doesn't seem to work) and to provide annotations API method, then update this PR.

**Comment by mmilata** [comment]
I've committed the required osbs changed so this is ready for review.

**Comment by TomasTomecek** [comment]
Works fine. Merging.

**Pull request merged**